externals/install-dotnet.sh

```diff
@@ -1346,6 +1386,8 @@ calculate_vars() {
     install_root="$(resolve_installation_path "$install_dir")"
     say_verbose "InstallRoot: '$install_root'."
 
+    normalized_architecture="$(get_normalized_architecture_for_specific_sdk_version "$version" "$normalized_channel" "$normalized_architecture")"
```

Comment:
In this place, the value of the `normalized_architecture` variable is mutated, it is better to get rid of this in order to avoid unexpected behavior in the future.